adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordRDDFunctions.scala

```diff
@@ -156,6 +158,16 @@ class AlignmentRecordRDDFunctions(rdd: RDD[AlignmentRecord])
           conf
         )
     }
+    if (asRegularFile) {
+      log.info("make regular BAM/SAM file (not hadoop)")
+      val conf = new Configuration()
+      val fs = FileSystem.get(conf)
+      val ouputParentDir = filePath.substring(0, filePath.lastIndexOf("/") + 1)
+      val tmpPath = ouputParentDir + "tmp" + System.currentTimeMillis().toString
+      fs.rename(new Path(filePath + "/part-r-00000"), new Path(tmpPath))
```

Comment:
A little digging implies that "old" (v1?) Hadoop wrote filenames like `part-00000`, cf. [this thread](http://hadoop-common.472056.n3.nabble.com/Is-it-always-called-part-00000-tp124567p124595.html) and [this Hadoop FAQ](https://wiki.apache.org/hadoop/FAQ#How_do_I_change_final_output_file_name_with_the_desired_name_rather_than_in_partitions_like_part-00000.2C_part-00001.3F).

[This SO](http://stackoverflow.com/a/10666874/544236) implies that we could end up with `part-m-00000` files if the output was from a map phase, which seems to never happen with Spark?

Since ADAM still supports Hadoop 1.0.4, we should maybe confirm that this does the right thing there and doesn't look for the wrong filename.